### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by DyaryRaoof**
- Added RSpec as a gem to your project.
- Created a Request spec file for all your controllers.
- Made sure that for each action you check:
  - If response status was correct.
  - If a correct template was rendered.
  - If the response body includes correct placeholder text.

### CODE REVIEW

File: spec/requests/posts_spec.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    end
+
+    it 'works! reponse body includes <h1>This is a list of posts<h1>' do
+      get user_posts_path(user_id: 1)
+      expect(response.body).to include('<h1>This is a list of posts<h1>')
+    end
+  end
+end
+
+RSpec.describe 'posts/:id', type: :request do
```
Comment (Dipeshtwis) [inline spec/requests/posts_spec.rb]: - [ ] I would recommend removing this `describe` from here to follow the DRY and rails principle.

File: spec/requests/users_spec.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    end
+
+    it 'works! reponse body includes <h1>This is a list of users<h1> ' do
+      get users_path
+      expect(response.body).to include('<h1>This is a list of users<h1>')
+    end
+  end
+end
+
+RSpec.describe 'users/:id', type: :request do
```
Comment (Dipeshtwis) [inline spec/requests/users_spec.rb]: - [ ] Here as well, I would recommend removing this `describe` block from here to follow the DRY and rails principle.

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
Hi @DyaryRaoof, 

:+1: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


 ## Required Changes ♻️

**_Check the comments under the review_**

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- N.A.

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**_Dipesh Kumar_**


------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/requests/posts_spec.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  it 'works!  renders show view correctly' do
+    get user_post_path(1, id: 1)
+    expect(response).to render_template(:show)
+  end
+
+  it 'works! reponse body includes <h1>this is a single post </h1> ' do
+    get user_post_path(1, id: 1)
+    expect(response.body).to include('<h1>this is a single post </h1>')
+  end
+end
```
Comment (Lameck1) [inline spec/requests/posts_spec.rb]: - [ ] Going back to the previous review concern. What was meant is that you should only have one **_`Rspec.describe`_** block per test file. What you should have done is take the test examples you have in the second **_`Rspec.describe`_** and move them first one. 

NOTE: It is allowed to have other **_describe_** blocks inside the main **_`Rspec.describe`_** block if separating contexts was the concern you were addressing.

File: spec/requests/users_spec.rb [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+  it 'works! renders show view correctly' do
+    get user_path(1)
+    expect(response).to render_template(:show)
+  end
+
+  it 'works! reponse body includes <h1>this is a single user </h1>' do
+    get user_path(1)
+    expect(response.body).to include('<h1>this is a single user </h1>')
+  end
```
Comment (Lameck1) [inline spec/requests/users_spec.rb]: - [ ] Please refactor this test file as discussed above so as to properly implement the changes requested by the first reviewer. Make the test file only have one **_`Rspec.describe`_** block.

**Review by Lameck1 — CHANGES_REQUESTED** [review]
Hi @DyaryRaoof, 

Great attempt addressing previous review concerns 👏. 
They are not properly addressed though. This new review attempts to explain what was meant by the previous reviewer.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Lameck1 — APPROVED** [review]
Hi, @DyaryRaoof,

You've done addressing previous review concerns 👏🏼 
Your project is now complete!  It's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**